nx/lib/nx.ex

```diff
@@ -7518,6 +7522,150 @@ defmodule Nx do
     )
   end
 
+  @doc """
+  Takes and concatenates slices along an axis.
+
+  Intuitively speaking, `take/3` reorders tensor slices along
+  the given axis based on the given indices, possibly duplicating
+  and removing slices.
+
+  Passing a multi-dimensional indices tensor only affects the
+  resulting shape. Specifically, the given axis in the input shape
+  gets replaced with the indices shape.
```

Comment:
What happens if a negative index is given?

Reply:
`BinaryBackend` uses slice so it clamps, but apparently so does xla::Gather. This isn't explicitly stated in the docs though. On the other hand `torch.index_select` raises an out of range error. Given the indices are dynamic this is pretty backend specific, do we want to impose some behaviour or leave it as undefined?

Reply:
Let’s raise in the binary backend then. If the other backends support it, it is fine.

Reply:
Sounds good!